integrations/spark/spark-validate-cleanse/spark-validate-cleanse-app/src/main/java/com/thinkbiganalytics/spark/datavalidator/Validator.java

```diff
@@ -45,57 +45,57 @@ public static void main(String[] args) {
+                         "You can optionally add: --numPartitions number_of_rdd_partitions\n" +
+                         "You provided " + args.length + " args which are (comma separated): " + StringUtils.join(args, ",");
+            out.println(msg);
+            throw new IllegalArgumentException(msg);
         }
 
-        final ValidatorConfiguration params = new ValidatorConfiguration(args);
-
-        // Initialize Spring context
-        final ApplicationContext ctx = new AnnotationConfigApplicationContext("com.thinkbiganalytics.spark");
-        final DataValidator app = ctx.getBean(DataValidator.class);
-
-        // Prepare Hive context
-        final HiveContext hiveContext = new HiveContext(SparkContext.getOrCreate());
-
-        for (final Param param : params.getHiveParams()) {
-            log.info("Adding Hive parameter {}={}", param.getName(), param.getValue());
-            hiveContext.setConf(param.getName(), param.getValue());
+        final SparkContext sparkContext = SparkContext.getOrCreate();
+        try {
+            final ValidatorConfiguration params = new ValidatorConfiguration(args);
+
+            // Initialize Spring context
+            final ApplicationContext ctx = new AnnotationConfigApplicationContext("com.thinkbiganalytics.spark");
+            final DataValidator app = ctx.getBean(DataValidator.class);
+
+            // Prepare Hive context
+            final HiveContext hiveContext = new HiveContext(sparkContext);
+
+            for (final Param param : params.getHiveParams()) {
+                log.info("Adding Hive parameter {}={}", param.getName(), param.getValue());
+                hiveContext.setConf(param.getName(), param.getValue());
+            }
+
+            log.info("Deployment Mode - {}", hiveContext.sparkContext().getConf().get("spark.submit.deployMode"));
+            Map<String, FieldPolicy> policyMap = ctx.getBean(FieldPolicyLoader.class).loadFieldPolicy(params.getFieldPolicyJsonPath());
+
+            // Run validation
+            final DataValidatorResult results = app.validateTable(params.getTargetDatabase(), params.getFeedTableName(), params.getValidTableName(), params.getPartition(), params.getNumPartitions(),
+                                                                  policyMap, hiveContext);
+
+            log.info("Persistence level: {}", params.getStorageLevel());
+            results.persist(StorageLevel.fromString(params.getStorageLevel()));
+
+            app.saveInvalidToTable(params.getTargetDatabase(), params.getInvalidTableName(), results, hiveContext);
+            app.saveValidToTable(params.getTargetDatabase(), params.getFeedTableName(), params.getValidTableName(), results, hiveContext);
+            app.saveProfileToTable(params.getTargetDatabase(), params.getProfileTableName(), params.getPartition(), results, hiveContext);
+            results.unpersist();
+            log.info("Validator app finished");
+            sparkContext.stop();
```

Comment:
Please move this to a `finally` block. It'll make the code cleaner and prevent bugs if new catch blocks are added.